velox/common/base/BitUtil.h

```diff
@@ -345,19 +346,35 @@ void forEachBit(
         if (!word) {
           return;
         }
-        while (word) {
-          func(idx * 64 + __builtin_ctzll(word));
-          word &= word - 1;
+        if ((kAllSet & mask) == (word & mask)) {
+          const size_t start = idx * 64 + __builtin_ctzll(word);
+          const size_t last = start + __builtin_popcountll(word & mask);
+          for (size_t row = start; row < last; ++row) {
+            func(row);
+          }
+        } else {
+          while (word) {
+            func(idx * 64 + __builtin_ctzll(word));
+            word &= word - 1;
+          }
         }
       },
       [isSet, bits, func](int32_t idx) {
         auto word = (isSet ? bits[idx] : ~bits[idx]);
         if (!word) {
           return;
         }
-        while (word) {
-          func(idx * 64 + __builtin_ctzll(word));
-          word &= word - 1;
+        if (kAllSet == bits[idx]) {
```

Comment:
Shouldn't this comparison use "word"? How would this work when the function is called with isSet == false? Looks like we might be missing some unit tests. Let's add.